server/komodo-rest/src/main/java/org/komodo/rest/service/KomodoDataserviceService.java

```diff
@@ -866,4 +870,265 @@ public Response getConnections( final @Context HttpHeaders headers,
+        if (principal.hasErrorResponse())
+            return principal.getErrorResponse();
+
+        List<MediaType> mediaTypes = headers.getAcceptableMediaTypes();
+        UnitOfWork uow = null;
+
+        try {
+            uow = createTransaction(principal, "getSourceVdbsForDataservice", true ); //$NON-NLS-1$
+
+            Dataservice dataservice = findDataservice(uow, dataserviceName);
+            if (dataservice == null)
+                return commitNoDataserviceFound(uow, mediaTypes, dataserviceName);
+
+            // Get JNDIs for the service physical sources
+            Vdb serviceVdb = dataservice.getServiceVdb(uow);
+            List<String> serviceJndis = new ArrayList<String>();
+            Model[] models = serviceVdb.getModels(uow);
+            for(Model model : models) {
+                if(model.getModelType(uow) == Model.Type.PHYSICAL) {
+                    ModelSource[] modelSources = model.getSources(uow);
+                    if(modelSources.length>0) {
+                        serviceJndis.add(modelSources[0].getJndiName(uow));
+                    }
+                }
+            }
+            
+            // Get list of source VDBs that have matching jndis
+            List<Vdb> sourceVdbs = new ArrayList<Vdb>();
+            for(String serviceJndi : serviceJndis) {
+                Vdb sourceVdb = getSourceVdbWithJndi(uow, serviceJndi);
+                if(sourceVdb!=null) sourceVdbs.add(sourceVdb);
+            }
+            
+            final List< RestVdb > entities = new ArrayList< >();
+            KomodoProperties properties = new KomodoProperties();
+            properties.addProperty(VDB_EXPORT_XML_PROPERTY, false);
+            for ( final Vdb vdb : sourceVdbs ) {
+                RestVdb entity = entityFactory.create(vdb, uriInfo.getBaseUri(), uow, properties);
+                entities.add(entity);
+            }
+
+            // create response
+            return commit( uow, mediaTypes, entities );
+
+        } catch ( final Exception e ) {
+            if ( ( uow != null ) && ( uow.getState() != State.ROLLED_BACK ) ) {
+                uow.rollback();
+            }
+
+            if ( e instanceof KomodoRestException ) {
+                throw ( KomodoRestException )e;
+            }
+
+            return createErrorResponseWithForbidden(mediaTypes, e, DATASERVICE_SERVICE_FIND_SOURCE_VDB_ERROR, dataserviceName);
+        }
+    }
+    
+    /*
+     * Returns first service source vdb that has source jndi matching the supplied jndi
+     */
+    private Vdb getSourceVdbWithJndi(final UnitOfWork uow, final String jndiName) throws KException {
+        // Look for source vdb that has physical source with matching jndi
+        WorkspaceManager wsMgr = getWorkspaceManager(uow);
+        Vdb[] wsVdbs = wsMgr.findVdbs(uow);
+        for(Vdb wsVdb : wsVdbs) {
+            if(isServiceSourceVdb(uow, wsVdb)) {
+                Model[] models = wsVdb.getModels(uow);
```

Comment:
Really small and optional comment. This loop would be easier to read if the first if-test was negated and continue used to skip the iteration. Reduces the number of braces.